media-api/test/lib/elasticsearch/ElasticSearchTestBase.scala

```diff
@@ -1,38 +1,61 @@
 package lib.elasticsearch
 
 import java.util.UUID
-import com.gu.mediaservice.lib.logging.{LogMarker, MarkerMap}
 import com.gu.mediaservice.model._
-import com.whisk.docker.impl.spotify.DockerKitSpotify
-import com.whisk.docker.scalatest.DockerTestKit
-import com.whisk.docker.{DockerContainer, DockerKit}
 import org.joda.time.DateTime
 import org.scalatest.concurrent.PatienceConfiguration.{Interval, Timeout}
 import org.scalatest.concurrent.ScalaFutures
 import org.scalatest.funspec.AnyFunSpec
 import org.scalatest.time.{Milliseconds, Seconds, Span}
 import org.scalatest.BeforeAndAfterAll
 import org.scalatest.matchers.should.Matchers
+import org.testcontainers.containers.wait.strategy.Wait
+import org.testcontainers.elasticsearch.ElasticsearchContainer
 import play.api.libs.json.JsString
 
-import scala.concurrent.duration._
+import java.util.concurrent.atomic.AtomicReference
 import scala.util.Properties
+import scala.concurrent.duration._
+import scala.compat.java8.DurationConverters._
+import scala.jdk.CollectionConverters._
 
-trait ElasticSearchTestBase extends AnyFunSpec with BeforeAndAfterAll with Matchers with ScalaFutures with Fixtures with DockerKit with DockerTestKit with DockerKitSpotify with ConditionFixtures {
 
+trait ElasticSearchTestBase extends AnyFunSpec with BeforeAndAfterAll with Matchers with ScalaFutures with Fixtures with ConditionFixtures {
 
   val interval = Interval(Span(100, Milliseconds))
   val timeout = Timeout(Span(10, Seconds))
 
   val useEsDocker = Properties.envOrElse("USE_DOCKER_FOR_TESTS", "true").toBoolean
-  val es6TestUrl = Properties.envOrElse("ES6_TEST_URL", "http://localhost:9200")
-
-  def esContainer: Option[DockerContainer]
-
-  final override def dockerContainers: List[DockerContainer] =
-    esContainer.toList ++ super.dockerContainers
 
-  final override val StartContainersTimeout = 1.minute
+  val esContainer: Option[ElasticsearchContainer] = if (useEsDocker) {
+    {
+      val container = new ElasticsearchContainer("docker.elastic.co/elasticsearch/elasticsearch:7.16.2")
+        .withExposedPorts(9200)
+        .withAccessToHost(true)
+        .withEnv(Map(
+          "cluster.name" -> "media-service",
+          "xpack.security.enabled" -> "false",
+          "discovery.type" -> "single-node",
+          "network.host" -> "0.0.0.0"
+        ).asJava)
+        .waitingFor(Wait.forHttp("/")
+          .forPort(9200)
+          .forStatusCode(200)
+          .withStartupTimeout(180.seconds.toJava)
+        )
+      container.start()
+      Some(container)
+    }
+  } else None
+
+  val esPort = esContainer.map(_.getMappedPort(9200)).getOrElse(9200)
+  val es6TestUrl = Properties.envOrElse("ES6_TEST_URL", s"http://localhost:$esPort")
+
+  override protected def afterAll(): Unit = {
+    super.afterAll()
+
+    esContainer foreach { _.stop() }
+  }
```

Comment:
any chance we could DRY this out and have this and thrall extend something in common? (appreciate it's not rule of 3, but seems like a lot of code with various config options therein which presents scope for divergence).